### PR TITLE
Make sure we extract and analyze en-gb as english correctly.

### LINKED ISSUE #6156: Extensions missing when searching with non-'en-US' locale
The `en-GB` locale (and other non-`en-US` locales) cannot find the Cookie AutoDelete extension.

https://addons.mozilla.org/en-GB/firefox/search/?q=cookie+autodelete&appver=57.0&platform=windows

https://addons.mozilla.org/en-US/firefox/search/?q=cookie+autodelete&appver=57.0&platform=windows

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Fixes #6156

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
Actually not yet ready for review, needs a valid search test-case to verify it's actually not just extracted but also can be found.

**Comment by EnTeQuAk** [comment]
@diox this should be ready for review. It essentially fixes one issue with extraction that en-gb content wasn't correctly extracted. It adds lots of tests verifying basic searches that are broken in #6156 (but aren't locally - I added the tests to prove to myself that I'm not missing something obvious) so I'd like to get this merged, tests are passing and potentially do a reindex on prod to get en-gb correctly extracted and check if #6156 is still happening.

File: src/olympia/constants/search.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
     'danish': ['da'],
     'german': ['de'],
     'greek': ['el'],
-    'english': ['en-us'],
+    'english': ['en-us', 'en-gb'],
```
Comment (diox) [inline src/olympia/constants/search.py]: So, wait... what happens when we try indexing an add-on with both an `en-us` and an `en-gb` translation?

Comment (EnTeQuAk) [inline src/olympia/constants/search.py]: the extracted strings are always extracted into a list so it'll look like that:

```
{
    'name_l10n_english': [ 'Banana Bacon (US)', 'Banana Bacon (GB)'']
    'name_l10n_spanish': ['Banana Bacon Espanol']
}
```

I wonder if we should make a special case for english to only index either but we do exactly the same for Chinese as well and I think it someone wants to have an explicit en-GB locale we should index it.

Comment (diox) [inline src/olympia/constants/search.py]: My main concern was when returning the strings back for the API but then I remembered it goes into a different, specific field that is not analyzed so it's fine :)

File: settings_test.py [hunk 2]
```diff
@@ -82,10 +83,10 @@ def _polite_tmpdir():
 # These are the default languages. If you want a constrainted set for your
 # tests, you should add those in the tests.
 AMO_LANGUAGES = (
```
Comment (diox) [inline settings_test.py]: Should we stop constraining the test languages btw? the list is huge now, I don't really see the point.

Comment (EnTeQuAk) [inline settings_test.py]: sounds like a great plan, I'll do that in a separate PR

**Pull request merged**